scripts/launch-s0-instance.sh

```diff
@@ -3,81 +3,233 @@
 me=$(basename $0)
 dir=$(readlink -f $(dirname $0))
 elastio_aws_id=537513441174
-security_group="elastio-s0-server"
+default_security_group="elastio-s0-server"
 default_instance_type="m5d.2xlarge"
 default_instance_name="$(whoami)-s0-server"
+default_block_size=fixed
+bootstrap=/tmp/s0-bootstrap.yml
+db_mount=/mnt/elastio
+unit_file=/etc/systemd/user/s0.service
+service=s0
+
+create_s0_bootstrap()
+{
+    rm -f $bootstrap
+    cat << EOF > $bootstrap
+#cloud-config
+# Bootstrap a 's0' server on Amazon Linux 2.
+# This script is intended to run as part of the cloud init
+
+write_files:
+  - path: /home/ec2-user/.aws/config
+    owner: ec2-user:ec2-user
+    content: |
+      [default]
+      output = json
+
+  - path: $UNIT_FILE
+    owner: ec2-user:ec2-user
+    permissions: '0664'
+    content: |
+      [Unit]
+      Description="Elastio Scale-to-Zero service"
+      After=network.target
+
+      [Service]
+      User=ec2-user
+      StartLimitInterval=5
+      StartLimitBurst=10
+      TimeoutStopSec=5min
+      ExecStart=/bin/bash -c '/usr/bin/s0 --log-output compact vault serve s3 --bucket $S0_BUCKET --key-id $S0_KMS $BLOCK_SIZE --server-address 0.0.0.0 --db-path $DB_MOUNT 2>&1 | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" >> /var/log/s0-service.log'
```

Comment:
the `key-id` parameter can be either the ugly AWS-generated key ID, or a human-readable alias.  If it's the latter, it needs the `alias/` prefix otherwise it will not be valid.  It would be great if you could do some minimal validation on the key ID before launching to avoid this failure mode.

Reply:
The validation makes sense. But I always used an alias without the mentioned prefix `alias/` and this command wasn't failing.
Where to put this prefix? Please explain how to use it. Do you mean something like this?
`--key-id alias/my-kms-key` ?

Reply:
It wasn't failing because of elastio/elastio#1327 but it would fail at runtime for sure.

Yes that is what I mean.  If you know that the parameter is always going to be an alias and never an actual key ID (and from the usage I'd say you know that), then you should verify it starts with `alias/` and if it doesn't you should prepend that prefix.

Reply:
Thanks for the response. It makes sense.